src/RequestHandlerSwooleRunner.php

```diff
@@ -193,24 +206,102 @@ function (Throwable $exception) use ($serverRequestErrorResponseGenerator) : Res
             ));
         }
 
-        $this->logger = $logger ?: new StdoutLogger();
+        $this->logger = $logger ? : new StdoutLogger();
+        $this->pidManager = $pidManager;
     }
 
     /**
      * Run the application
      */
     public function run() : void
     {
-        $this->swooleHttpServer->on('start', [$this, 'onStart']);
-        $this->swooleHttpServer->on('request', [$this, 'onRequest']);
-        $this->swooleHttpServer->start();
+        $opts = new Getopt([
+            'daemonize|d'  => 'daemonize the swoole server process',
+            'worker_num|n=i' => 'set worker nums',
+            'dispatch_mode=i' => 'set dispatch_mode',
+        ]);
+        $args = $opts->getArguments();
+        $action = $args[0] ?? null;
+        switch ($action) {
+            case 'stop':
+                $this->stopServer();
+                break;
+            case 'start':
+            default:
+                $this->startServer([
+                    'daemonize' => $opts->getOption('daemonize') ? (bool)$opts->getOption('daemonize') : false,
```

Comment:
Here and elsewhere: when casting, put a space between the cast and the variable: `(bool) $opts->getOption(/*...*/)`.